tests/test_gui_layout.py

```
@@ -0,0 +1,193 @@
+        config.save = lambda data: None
+        tray.AVAILABLE = False              # no tray icon during the test
+
+        from launcher import gui
+        launcher_main._apply_gui_review_fixes(gui)
+        cls.gui = gui
+        cls.tk = tk
+
+        try:
+            cls.root = tk.Tk()
+        except tk.TclError:
+            raise unittest.SkipTest("no display")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)